app/src/main/java/org/mozilla/fenix/browser/readermode/ReaderModeController.kt

```diff
@@ -0,0 +1,25 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.browser.readermode
+
+import mozilla.components.feature.readerview.ReaderViewFeature
+import mozilla.components.support.base.feature.ViewBoundFeatureWrapper
+
+interface ReaderModeController {
+    fun hideReaderView()
+    fun showReaderView()
+}
+
+class DefaultReaderModeController(
+    private val readerViewFeature: ViewBoundFeatureWrapper<ReaderViewFeature>
+) : ReaderModeController {
+    override fun hideReaderView() {
+        readerViewFeature.withFeature { it.hideControls() }
```

Comment:
Ah there's a bug here :). This seems to mix up `showReaderView`/`hideReaderView` and `showControls`/`hideControls`. The former is for activating and hiding reader mode, the latter for the appearance controls.

Reply:
So, in other words this should be `hideReaderView` not `hideControls`.

Reply:
Good catch!